tests/test_actor.py

```diff
@@ -526,45 +613,52 @@ def test_pricing(self):
         assert order.price == sell_price
 
+    def test_geometric_pricing(self):
         # test geometric series
         # this series has a constant factor in between the elements of the series, so that the
         # n-th element, with the geometric factor x and the start value n0 is defined as
         # n-th element = n0 * x^(n)
-        actor.pricing_strategy = dict(name="geometric", param=[0.9])
+        buy_price = 10
+        sell_price = 1
+        actor = self.get_actor_w_pricing_setup(capacity=10, soc_initial=0.5,
+                                               buy_price=buy_price, sell_price=sell_price)
+        GEOMETRIC_FACTOR = 0.9
+        actor.pricing_strategy = dict(name="geometric", param=[GEOMETRIC_FACTOR])
         check_index = 3
         actor.market_schedule[:] = 0
         actor.market_schedule[check_index] = 1
         order = actor.generate_order()
-        assert order.price == pytest.approx(buy_price*0.9**3)
+        assert order.price == pytest.approx(buy_price*GEOMETRIC_FACTOR**check_index)
 
         actor.market_schedule[:] = 0
         actor.market_schedule[check_index] = -1
         order = actor.generate_order()
-        assert order.price == pytest.approx(sell_price * (1/0.9) ** 3)
+        assert order.price == pytest.approx(sell_price * (1/GEOMETRIC_FACTOR) ** check_index)
 
         # a second parameter can be used to cap the price to a factor of the final price
         # the pricing is capped/clipped at a price of factor*final_price, in this case 70% of the
         # final price- the upper bound is symmetric so that it is bound to 1/0.7 --> 143% of the
         # final price
-        actor.pricing_strategy = dict(name="geometric", param=[0.9, 0.7])
+        SYMMETRIC_BOUND = 0.7
+        actor.pricing_strategy = dict(name="geometric", param=[GEOMETRIC_FACTOR, SYMMETRIC_BOUND])
         check_index = 6
         actor.market_schedule[:] = 0
         actor.market_schedule[check_index] = 1
         order = actor.generate_order()
-        assert order.price == 0.7*buy_price
+        assert order.price == SYMMETRIC_BOUND*buy_price
 
         actor.market_schedule[:] = 0
         actor.market_schedule[check_index] = -1
         order = actor.generate_order()
-        assert order.price == pytest.approx(1/0.7 * sell_price)
+        assert order.price == pytest.approx(1/SYMMETRIC_BOUND * sell_price)
 
     def test_market_schedule_adjustment(self):
         # test if the market schedule is properly adjusted when orders are matched with the pricing
         # strategy.
         battery = Battery(capacity=10, max_c_rate=2, soc_initial=0.0)
-        # with this soc and capacity it means max 0.1 energy can be stored
+        # with this soc and capacity it means max 10 energy can be stored
         actor = Actor(0, self.example_df, battery=battery, _steps_per_hour=4, cluster=0)
-        actor.pricing_strategy = dict(name="linear", param=[0.1])
+        actor.pricing_strategy = dict(name="linear", param=[0])
```

Comment:
A change in the linear pricing led to negative order generation, which led to no matching and a failing test. Since this test is not about the pricing strategy, param=0 removes an error source for this test and makes it more specific for its designed purpose